src/test/java/guitests/PersonEditGuiTest.java

```diff
@@ -139,22 +136,45 @@ public void cancelOperation_usingAccelerator() {
         Person pandaWong = new PersonBuilder("Panda", "Wong")
                 .withStreet("Chengdu Panda Street").withCity("Chengdu").withPostalCode("PANDA")
                 .withBirthday("01.01.1979").withGithubUsername("panda").withTags(td.colleagues, td.friends).build();
-        addPersonDialog.enterNewValues(pandaWong);
-        addPersonDialog.clickOk();
-
-        personListPanel.clickOnListView(); // To ensure shortcut keys work properly.
+        addPersonDialog.enterNewValues(pandaWong).clickOk();
 
         personListPanel.use_LIST_GOTO_BOTTOM_SEQUENCE();
         PersonCardHandle pandaWongCardHandle = personListPanel.getPersonCardHandle(pandaWong);
-        assertEquals(pandaWongCardHandle, pandaWong);
+        assertEquals(pandaWongCardHandle, pandaWong); //Ensure correct state before cancelling.
+
         personListPanel.use_PERSON_CHANGE_CANCEL_ACCELERATOR();
         assertNull(personListPanel.getPersonCardHandle(pandaWong));
+    }
+
+    @Test
+    public void editPerson_editDuringGracePeriod() {
+        Person newAlice = new PersonBuilder(td.alice.copy()).withFirstName("Alicia").withLastName("Brownstone")
+                .withStreet("Updated street").withCity("Singapore").withPostalCode("123123")
+                .withBirthday("01.01.1979").withGithubUsername("alicebrown123").withTags(td.colleagues, td.friends).build();
 
-        //Ensure cancel operation has no side effects after grace period.
+        //Get a reference to the card displaying Alice's details
+        PersonCardHandle alicePersonCard = personListPanel.selectCard(td.alice);
+
+        //Edit Alice to change to new values
+        EditPersonDialogHandle editPersonDialog = personListPanel.use_PERSON_EDIT_ACCELERATOR();
+        editPersonDialog.enterNewValues(newAlice).pressEnter();
+
```

Comment:
Yes, no harm doing again. Sometimes things work the first time but not the second time.